sled-storage/src/manager.rs

```diff
@@ -2005,16 +1985,39 @@ mod tests {
             .expect("Ensuring disks should work after key manager is ready");
         assert!(!result.has_error(), "{:?}", result);
 
-        // Create a filesystem on the newly formatted U.2
-        let dataset_id = Uuid::new_v4();
+        // Create a filesystem on the newly formatted U.2.
+        //
+        // We can call "upsert_filesystem" both with and without a UUID.
```

Comment:
Would it make sense to add a test for calling with `None` and seeing the dataset come back without an ID?

Reply:
Sure, added a new test: `upsert_filesystem_no_uuid`